src/Http/Url.php

```diff
@@ -68,9 +68,8 @@ public function match($patterns)
             $patterns = [$patterns];
         }
 
-        //Remove scheme and query
-        $url = preg_replace('|(\?.*)?$|', '', (string) $this);
-        $url = preg_replace('|^(\w+://)|', '', $url);
+        //Remove scheme
+        $url = preg_replace('|^(\w+://)|', '', (string) $this);
```

Comment:
Nice workaround, but it also makes it impossible to require the parameter to start with either `?` or `&` in a single string. I've fixed that by "doubling" the blacklist, one string for each.